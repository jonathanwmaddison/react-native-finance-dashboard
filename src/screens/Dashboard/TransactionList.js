import React, { Component } from 'react';
import { connect } from 'react-redux';
import { shape, number, string, arrayOf } from 'prop-types';
import { View, FlatList, Text } from 'react-native';
import TransactionItem from './TransactionItem';
import { getTransactionsState } from './selectors';

export class TransactionList extends Component {
  static propTypes = {
    transactions: arrayOf(
      shape({
        transId: number.isRequired,
        custId: number.isRequired,
        transTime: number.isRequired,
        description: string.isRequired,
        transTo: string.isRequired,
        transFrom: string.isRequired,
      }).isRequired
    ),
  };

  keyExtractor = (item) => item.transId.toString();

  render() {
    const { transactions } = this.props;
    return (
      <View style={{ flex: 1}}>
        <FlatList
          data={transactions}
          keyExtractor={this.keyExtractor}
          renderItem={({ item }) => (
            <TransactionItem
              {...item}
            />
          )}
        />
        </View>
    );
  }
}
const mapStateToProps = (state, props) => {
  return {
    transactions: getTransactionsState(state, props)
  }
};

export default connect(mapStateToProps)(TransactionList);
