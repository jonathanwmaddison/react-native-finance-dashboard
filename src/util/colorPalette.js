export default {
  white: '#e9e4e5',
  green: '#a8a766',
  blue: '#529dbe',
  navy: '#2c6a88',
  purple: '#352739'
};
