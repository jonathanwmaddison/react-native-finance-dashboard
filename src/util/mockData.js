const mockData = [
  {
    "transId": 24,
    "custId": 0,
    "transTime": 1515703912733,
    "transAmt": -53.23,
    "description": "Team Lunch",
    "transTo": "Restaurant",
    "transFrom": "Checking Account"
  },
  {
    "transId": 23,
    "custId": 0,
    "transTime": 1515703912113,
    "transAmt": -53.23,
    "description": "Team Lunch",
    "transTo": "Restaurant",
    "transFrom": "Checking Account"
  },
  {
    "transId": 22,
    "custId": 0,
    "transTime": 1515703911630,
    "transAmt": -53.23,
    "description": "Team Lunch",
    "transTo": "Restaurant",
    "transFrom": "Checking Account"
  },
  {
    "transId": 21,
    "custId": 0,
    "transTime": 1515703910985,
    "transAmt": -53.23,
    "description": "Team Lunch",
    "transTo": "Restaurant",
    "transFrom": "Checking Account"
  },
  {
    "transId": 20,
    "custId": 0,
    "transTime": 1515703910265,
    "transAmt": -53.23,
    "description": "Team Lunch",
    "transTo": "Restaurant",
    "transFrom": "Checking Account"
  },
  {
    "transId": 19,
    "custId": 0,
    "transTime": 1515703909606,
    "transAmt": -53.23,
    "description": "Team Lunch",
    "transTo": "Restaurant",
    "transFrom": "Checking Account"
  },
  {
    "transId": 18,
    "custId": 0,
    "transTime": 1515703909120,
    "transAmt": -53.23,
    "description": "Team Lunch",
    "transTo": "Restaurant",
    "transFrom": "Checking Account"
  },
  {
    "transId": 17,
    "custId": 0,
    "transTime": 1515703868742,
    "transAmt": 621.23,
    "description": "Profit",
    "transTo": "Checking Account",
    "transFrom": "Customers"
  },
  {
    "transId": 16,
    "custId": 0,
    "transTime": 1515703868006,
    "transAmt": 621.23,
    "description": "Profit",
    "transTo": "Checking Account",
    "transFrom": "Customers"
  },
  {
    "transId": 15,
    "custId": 0,
    "transTime": 1515703866928,
    "transAmt": 621.23,
    "description": "Profit",
    "transTo": "Checking Account",
    "transFrom": "Customers"
  },
  {
    "transId": 14,
    "custId": 0,
    "transTime": 1515703018512,
    "transAmt": 621.23,
    "description": "Profit",
    "transTo": "Checking Account",
    "transFrom": "Customers"
  },
  {
    "transId": 13,
    "custId": 0,
    "transTime": 1515703016968,
    "transAmt": 621.23,
    "description": "Profit",
    "transTo": "Checking Account",
    "transFrom": "Customers"
  },
  {
    "transId": 12,
    "custId": 0,
    "transTime": 1515703011771,
    "transAmt": 621.23,
    "description": "Profit",
    "transTo": "Checking Account",
    "transFrom": "Customers"
  },
  {
    "transId": 11,
    "custId": 0,
    "transTime": 1515702996305,
    "transAmt": 421,
    "description": "Profit",
    "transTo": "Checking Account",
    "transFrom": "Customers"
  },
  {
    "transId": 10,
    "custId": 0,
    "transTime": 1515702985680,
    "transAmt": 20000,
    "description": "Profit",
    "transTo": "Checking Account",
    "transFrom": "Customers"
  },
  {
    "transId": 9,
    "custId": 0,
    "transTime": 1515702966650,
    "transAmt": -20000,
    "description": "Savings",
    "transTo": "Savings Account",
    "transFrom": "Checking Account"
  },
  {
    "transId": 8,
    "custId": 0,
    "transTime": 1515702798901,
    "transAmt": -33000.13,
    "description": "Salaries",
    "transTo": "Employees",
    "transFrom": "Checking Account"
  },
  {
    "transId": 7,
    "custId": 0,
    "transTime": 1515702781539,
    "transAmt": -1000.13,
    "description": "Macbook air pro",
    "transTo": "Apple",
    "transFrom": "Checking Account"
  },
  {
    "transId": 6,
    "custId": 0,
    "transTime": 1515701468425,
    "transAmt": -3232.13,
    "description": "Macbook pro",
    "transTo": "Apple",
    "transFrom": "Checking Account"
  },
  {
    "transId": 5,
    "custId": 0,
    "transTime": 1515701467522,
    "transAmt": -3232.13,
    "description": "Macbook pro",
    "transTo": "Apple",
    "transFrom": "Checking Account"
  },
  {
    "transId": 4,
    "custId": 0,
    "transTime": 1515701461996,
    "transAmt": -3000,
    "description": "Macbook pro",
    "transTo": "Apple",
    "transFrom": "Checking Account"
  },
  {
    "transId": 3,
    "custId": 0,
    "transTime": 1515701443154,
    "transAmt": -2000,
    "description": "Hamilton tickets",
    "transTo": "Lin Manuel Miranda",
    "transFrom": "Checking Account"
  },
  {
    "transId": 2,
    "custId": 0,
    "transTime": 1515701406019,
    "transAmt": 500000,
    "description": "VC money",
    "transTo": "Checking Account",
    "transFrom": "VC 2"
  },
  {
    "transId": 1,
    "custId": 0,
    "transTime": 1515701357307,
    "transAmt": 50000,
    "description": "Initial deposit",
    "transTo": "Checking Account",
    "transFrom": "New VC"
  }
];
