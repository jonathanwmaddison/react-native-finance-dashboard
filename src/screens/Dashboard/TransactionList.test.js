import React from 'react';
import renderer from 'react-test-renderer'
import { mockTransactions } from '../../util/mockData';
import { TransactionList } from './TransactionList'

jest.mock('./TransactionItem', () => 'TransactionItem')

describe('TransactionList', () => {
  let wrapper;
  let props = {
    sortByKey: 'transAmt',
    search: null,
    ascendingSort: false
  };
  beforeEach(() => {
    wrapper = renderer
      .create(<TransactionList {...props} transactions={mockTransactions} />)
  });
  it('[snapshot] matches', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('[method] keyExtractor gets correct key', () => {
    expect(wrapper.getInstance().keyExtractor(mockTransactions[0])).toBe('24')
  })
})
