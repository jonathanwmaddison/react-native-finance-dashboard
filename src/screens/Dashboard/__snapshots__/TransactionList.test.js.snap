// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionList [snapshot] matches 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "custId": 0,
          "description": "Team Lunch",
          "transAmt": -53.23,
          "transFrom": "Checking Account",
          "transId": 24,
          "transTime": 1515703912733,
          "transTo": "Restaurant",
        },
        Object {
          "custId": 0,
          "description": "Team Lunch",
          "transAmt": -53.23,
          "transFrom": "Checking Account",
          "transId": 23,
          "transTime": 1515703912113,
          "transTo": "Restaurant",
        },
        Object {
          "custId": 0,
          "description": "Team Lunch",
          "transAmt": -53.23,
          "transFrom": "Checking Account",
          "transId": 22,
          "transTime": 1515703911630,
          "transTo": "Restaurant",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <TransactionItem
          custId={0}
          description="Team Lunch"
          transAmt={-53.23}
          transFrom="Checking Account"
          transId={24}
          transTime={1515703912733}
          transTo="Restaurant"
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <TransactionItem
          custId={0}
          description="Team Lunch"
          transAmt={-53.23}
          transFrom="Checking Account"
          transId={23}
          transTime={1515703912113}
          transTo="Restaurant"
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <TransactionItem
          custId={0}
          description="Team Lunch"
          transAmt={-53.23}
          transFrom="Checking Account"
          transId={22}
          transTime={1515703911630}
          transTo="Restaurant"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
